### PR TITLE
BGL parametrisation of the B->D and B->D^* form factors

### PR DESCRIPTION
**Pull request by dvandyk**
Contributed by @cBobeth and @gubernari.

@cBobeth I have kept parts of the changes to B->Vlnu, which were still necessary for finite results at the endpoint even with your switch from A_2 -> A_12.

### CODE REVIEW

File: eos/statistics/log-likelihood.cc [hunk 1]
```diff
@@ -1330,7 +1330,7 @@ namespace eos
                 {
                     double llh = (*b)->evaluate();
                     if (! std::isfinite(llh))
-                        return llh;
+                        return -std::numeric_limits<double>::infinity();
```
Comment (cBobeth) [inline eos/statistics/log-likelihood.cc:1333]: Is the minus sign in front of std::... ok?

File: eos/parameters/hadronic-matrix-elements.yaml [hunk 2]
```diff
@@ -1096,6 +1169,103 @@ groups:
           min:     +4.667079193615467
           max:     +4.667079193615467
           comments: ''
+      # form factor parameters for B->D for [BGL1997]
```
Comment (cBobeth) [inline eos/parameters/hadronic-matrix-elements.yaml:1172]: Correct in comment: ... B->D^* ...

**Comment by cBobeth** [comment]
For me the canges seem fine.

File: eos/statistics/log-likelihood.cc (hunk 1 above)
Comment (dvandyk) [reply eos/statistics/log-likelihood.cc:1333]: Yes.

**Comment by MartinJung1978** [comment]
Hi, remember that one of the BGL parameters can be eliminated because of form-factor relations at q2=0. Also, the fit cannot give the overall normalization, since we only consider shape information.

**Comment by MartinJung1978** [comment]
I had a look at the tests: they seem ok, i.e. I get similar values. I haven't tried to reproduce yours exactly, so probably some inputs are slightly different at the moment.

**Comment by dvandyk** [comment]
> Hi, remember that one of the BGL parameters can be eliminated because of form-factor relations at q2=0. Also, the fit cannot give the overall normalization, since we only consider shape information.

@MartinJung1978 I agree that this should be implemented. However, the current implementation is not wrong in that regard. Hence, let's merge this first, and improve later.

File: eos/parameters/hadronic-matrix-elements.yaml (hunk 2 above)
Comment (dvandyk) [reply eos/parameters/hadronic-matrix-elements.yaml:1172]: Fixed.

**Pull request merged**